hibernate-core/src/main/java/org/hibernate/query/spi/AbstractSelectionQuery.java

```diff
@@ -181,16 +181,27 @@ protected void beforeQuery() {
 
 		session.prepareForQueryExecution( requiresTxn( options.getLockOptions().getLockMode() ) );
 		prepareForExecution();
+		prepareSessionFlushMode( session );
+		prepareSessionCacheMode( session );
+	}
 
+	/*
+	 * Used by Hibernate Reactive
+	 */
+	protected void prepareSessionFlushMode(SharedSessionContractImplementor session) {
 		assert sessionFlushMode == null;
-		assert sessionCacheMode == null;
-
-		final FlushMode effectiveFlushMode = getQueryOptions().getFlushMode();
+		final var effectiveFlushMode = getQueryOptions().getFlushMode();
 		if ( effectiveFlushMode != null && session instanceof SessionImplementor statefulSession ) {
 			sessionFlushMode = statefulSession.getHibernateFlushMode();
 			statefulSession.setHibernateFlushMode( effectiveFlushMode );
 		}
+	}
 
+	/*
+	 * Used by Hibernate Reactive
+	 */
+	protected void prepareSessionCacheMode(SharedSessionContractImplementor session) {
+		assert sessionCacheMode == null;
 		final CacheMode effectiveCacheMode = getCacheMode();
```

Comment:
For consistency with the change above.

```suggestion
		final var effectiveCacheMode = getCacheMode();
```